searchlib/src/vespa/searchlib/attribute/searchcontextelementiterator.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include "searchcontextelementiterator.h"
+#include <vespa/searchcommon/attribute/i_search_context.h>
+#include <vespa/searchlib/fef/termfieldmatchdata.h>
+
+using search::fef::TermFieldMatchDataPosition;
+
+namespace search::attribute {
+
+void
+SearchContextElementIterator::getElementIds(uint32_t docId, std::vector<uint32_t> & elementIds) {
+    int32_t weight(0);
+    for (int32_t id = _searchContext.find(docId, 0, weight); id >= 0; id = _searchContext.find(docId, id+1, weight)) {
+        elementIds.push_back(id);
+    }
+}
+void
+SearchContextElementIterator::mergeElementIds(uint32_t docId, std::vector<uint32_t> & elementIds) {
+    size_t toKeep(0);
+    int32_t id(-1);
+    int32_t weight(0);
+    for (int32_t candidate : elementIds) {
+        if (candidate >= id) {
+            id = _searchContext.find(docId, candidate, weight);
```

Comment:
possible re-write: avoid calling find multiple times for same candidate
possible re-write: terminate loop when attribute matching is exhausted.

Reply:
Fixed